src/js/defaulttheme.stories.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { base } from 'grommet/themes';
+import { deepMerge } from 'grommet/utils';
+
+import { Box, themeDefaultProps } from 'grommet';
+
+themeDefaultProps.theme = deepMerge(base, {
```

Comment:
But, this will be a published interface. So, is the guidance going to be that you should set this before using any Grommet components and not change this after any components have been used?

If so, I'm wondering if we should deprecate `theme` in the Grommet component and use this as the preferred means of customizing the Grommet theme?

Also, since this is very theme centric, and we don't anticipate any other defaultProps customizations, I'm wondering if we should do this in a more theme-specific way? For example, have something like `Grommet.initializeTheme()` that takes a theme and sets up whatever we need internally. That seems like a more caller-centric way of approaching it, whereas the defaultProps way feels more like our internal way of thinking about it.

Reply:
I think the idea here as far as interface/API goes is for those wrapping Grommet and not a general interface. I like the idea of `initializeTheme` which takes in a theme object and sets the default props under the hood.

Reply:
agreed. I added `initializeDefaultTheme`. i like `default` since it is a clear indication that will replace the default theme.

Reply:
Minor suggestion, change to `initializeDefaults` and allow any defaults to be passed. While we know theme is a safe default right now, a system may want to add others.

```js
import { initializeDefaults } from 'grommet';
const theme = { /* my theme */ };

initializeDefaults({ theme });
```

Reply:
Can you provide an example of anything else we might want to handle in this path? I ask because we've never had any other "global" props other than theme so far.

Reply:
Shouldn't that object that is passed be set as the prototype for all components? I could see someone wanting to set a default `allyTitle` or even setting some default props you want to have access to in your style prop functions.

Reply:
hum I see. let my play with this locally.

Reply:
I agree this feels more flexible, and future proof

Reply:
Why would I want a default `a11yTitle` for all components?

Reply:
@ericsoderberghp contrived example: a system wrapping Grommet needs `allyTitle` to be a required field for accessibility reasons. Something like setting a unique token `allyTitle: 'MISSING_ALLY_TITLE'` that can be scanned for and reported if missing. I've seen this done for internationalization to ensure no strings are hardcoded.

As this feature is for a different audience than the common user consuming Grommet components as-is, I'm not sure it makes sense to limit the scope to `theme`. I would trust the user knows what they are doing if they are using this feature.